### PR TITLE
fix: generate pretty package version name properly

### LINKED ISSUE #14838: Commit messages have an extra `v` before the version with custom versioning
An interesting side effect of this, PRs have an extra `v` before the tag name.
`chore(deps): update dependency our-org/monorepo to vfoo-pkg-v3.2.1`

_Originally posted by @ChipWolf in https://github.com/renovatebot/renovate/discussions/14759#discussioncomment-2456866_

### PR DESCRIPTION
**Pull request by pret-a-porter**
## Changes

Fix generating pretty package version name

## Context

- closes https://github.com/renovatebot/renovate/issues/14838

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/updates/generate.ts [hunk 1]
```diff
@@ -84,9 +84,9 @@ export function generateBranchConfig(
     toValues.add(upg.newValue!);
     // prettify newVersion and newMajor for printing
     if (upg.newVersion) {
-      upg.prettyNewVersion = upg.newVersion.startsWith('v')
-        ? upg.newVersion
-        : `v${upg.newVersion}`;
+      upg.prettyNewVersion = semver.valid(upg.newVersion)
```
Comment (viceice) [inline lib/workers/repository/updates/generate.ts:87]: not sure if we should only do this for semver compliant values. 🤔

Comment (pret-a-porter) [reply lib/workers/repository/updates/generate.ts:87]: Me too, but for now, all unit test cases are related to semantic versioning

Comment (rarkins) [reply lib/workers/repository/updates/generate.ts:87]: Can we use the selected versioning for the upgrade plus maybe a check that it starts with a number?

**Review by viceice — CHANGES_REQUESTED** [review]
needs deconfliction

File: lib/workers/repository/updates/generate.ts (hunk 1 above)
Comment (pret-a-porter) [reply lib/workers/repository/updates/generate.ts]: Added function `prettifyVersion`

File: lib/workers/repository/updates/generate.ts [hunk 2]
```diff
@@ -12,6 +12,14 @@ import * as template from '../../../util/template';
 import type { BranchConfig, BranchUpgradeConfig } from '../../types';
 import { CommitMessage } from '../model/commit-message';
 
+function prettifyVersion(version: string | number): string {
+  if (regEx(/^\d/).test(String(version))) {
```
Comment (viceice) [inline lib/workers/repository/updates/generate.ts:16]: i think we can remove number handling, see other comment

File: lib/workers/repository/updates/generate.ts [hunk 3]
```diff
@@ -84,12 +92,10 @@ export function generateBranchConfig(
...
     // prettify newVersion and newMajor for printing
     if (upg.newVersion) {
-      upg.prettyNewVersion = upg.newVersion.startsWith('v')
-        ? upg.newVersion
-        : `v${upg.newVersion}`;
+      upg.prettyNewVersion = prettifyVersion(upg.newVersion);
     }
     if (upg.newMajor) {
-      upg.prettyNewMajor = `v${upg.newMajor}`;
+      upg.prettyNewMajor = prettifyVersion(upg.newMajor);
```
Comment (viceice) [inline lib/workers/repository/updates/generate.ts:98]: i don't think we need this change, as major is always a number.

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**